### ISSUE TITLE
[func-call-spacing] False positive when declaring method in type parameter interface

### ISSUE THREAD

**Issue by gnuletik** [typescript-eslint/typescript-eslint#4299]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/main/docs/linting/TROUBLESHOOTING.md) and my problem is not listed.

```ts
/*eslint func-call-spacing: ["error", "never"]*/

// throws error
const emit = defineEmits<{
  (e): void
}>()

// works
interface Emits {
  (e): void
}

const emit2 = defineEmits<Emits>()
```

https://yeonjuan.github.io/typescript-eslint-demo/#e2NvZGU6Jy8qZXNsaW50IGZ1bmMtY2FsbC1zcGFjaW5nOiBbImVycm9yIiwgIm5ldmVyIl0qL1xuXG4vLyBkb2VzblwndCB3b3JrXG5jb25zdCBlbWl0ID0gZGVmaW5lRW1pdHM8e1xuICAoZSk6IHZvaWRcbn0+KClcblxuLy8gd29ya3NcbmludGVyZmFjZSBFbWl0cyB7XG5cdChlKTogdm9pZFxufVxuXG5jb25zdCBlbWl0MiA9IGRlZmluZUVtaXRzPEVtaXRzPigpJ30=

**Expected Result**

Doesn't throw error.

**Actual Result**

Throws error :
```
4:27 - Unexpected whitespace between function name and paren.(func-call-spacing)
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.26.0` |
| `@typescript-eslint/parser`        | `4.2.6.0` |
| `TypeScript`                       | `4.3` |
| `ESLint`                           | `7` |
| `node`                             | `16` |

**Comment by bradzacher** [comment]
https://typescript-eslint.io/docs/linting/troubleshooting#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code

**Comment by gnuletik** [comment]
Oh my bad!
Sorry for the noise and thanks for the help!
